ios/Classes/AirshipInboxMessageView.swift

```diff
@@ -79,8 +79,24 @@ class AirshipInboxMessageView : NSObject, FlutterPlatformView, NativeBridgeDeleg
                                 details: nil))
             return
         }
-        
+
         let inbox = MessageCenter.shared.inbox
+
+        let message = await inbox.message(forID: messageId)
+
+        if message == nil {
+            /// Attempt a refresh is the message isn't available - as can happen when launched from a push
+            let success = try? await inbox.refreshMessages(timeout: 5)
```

Comment:
Should we just wait forever if it fails ? What would the handling look like on the flutter side for this?

Reply:
yeah probably - the async call just never receives a result, which gives the user the chance on the flutter side to continue showing a loading dialog